src/filters/filter_lua_script.cpp

```diff
@@ -0,0 +1,382 @@
+            }
+            in_color.at(i) = input_channels.color_channels.at(names_in_color.at(i));
+            // Todo delete in_color;
+            lua[names_in_color.at(i)] = std::ref(input_channels.color_channels.at(names_in_color.at(i)));
+        }
+
+        if (!initial_parameters.contains("script")) {
+            throw filter_config_exception("lua filter: unable to setup the script");
+        }
+
+//        if (!initial_parameters.contains("patching")) {
+//            throw filter_config_exception("lua filter: unable to setup the patching");
+//        }
+
+        // initial_parameters.at("patching") divide
+        // for each fixture
+        //      fix_univ
+        //      fix_name
+        //      fix_ch_size
+        //      fixture local = fixture(fix_univ, fix_name, fix_ch_size);
+        //      ? dynamisch verschieden viele Argumente übergeben?
+//              lua.new_usertype<fixture>("Fixture", "pan", &fixture.channel_values.at(0), "s", &dmxfish::dmx::pixel::saturation, "i", &dmxfish::dmx::pixel::iluminance);
+        //      fixtures.append(local);
+
+
+
+        lua.open_libraries(sol::lib::base, sol::lib::package);
+        lua.open_libraries(sol::lib::math, sol::lib::table, sol::lib::string);
+        lua.set_function("update", []() {
+        });
+        lua.set_function("scene_activated", []() {
+        });
+
+//        for(fixture fix: fixtures){
+//            lua.create_named_table(fix.name, "universe", fix.universe,
+//                                   "first_channel", fix.first_channel);
+//        }
+        lua.create_named_table("output");
+
+        lua.set_function( "hsi_to_rgb", sol::overload(
+                hsi_to_rgb_color,
+                hsi_to_rgb_table
+        ) );
+
+        lua.set_function( "hsi_to_rgbw", sol::overload(
+                hsi_to_rgbw_color,
+                hsi_to_rgbw_table
+        ) );
+
+        try {
+            lua.script(initial_parameters.at("script"));
+        } catch (std::exception &e){
+            throw filter_config_exception(std::string("setup the filter threw an error: ") + e.what());
+        }
+
+        sol::object scene_activated_obj = lua["scene_activated"];
+        if (scene_activated_obj.get_type() == sol::type::function && scene_activated_obj.is<std::function<void()>>()){
+            scene_activated_lua = scene_activated_obj;
+        } else {
+            throw filter_config_exception("scene_activated is not a function or has the wrong signature");
+        }
+
+        sol::object update_obj = lua["update"];
+        if (update_obj.get_type() == sol::type::function && update_obj.is<std::function<void()>>()){
+            update_lua = update_obj;
+        } else {
+            throw filter_config_exception("update is not a function or has the wrong signature");
+        }
+    }
+
+    bool filter_lua_script::receive_update_from_gui(const std::string &key, const std::string &_value) {
+
```

Comment:
We should also check if we can forward the k,v pair to the script and return the processing result from the script. If the corresponding method isn't implemented we should return false right away.

Reply:
a key value pair for each method, set by the user? Is it not enough to throw an error if something in update goes wrong? which type of exception can the caller of update handle?

Reply:
No, you got me wrong. We should do something like this pseudo-code:
```
bool receive_update_from_gui(k, v):
    if not lua_context.has_function("receive_gui_update"):
        return false
    else:
        try:
            return (bool) lua_context.exec_function("receive_gui_update", k, v)
        catch:
            return false
```

Reply:
should work right now